persist/fs/write.go

```diff
@@ -21,14 +21,18 @@
 package fs
```

Comment:
There is a lot going on in this file. I count 6 different file descriptors. Seems like generally speaking now (or sometime soon with the subsequent P.R's) would be a good time to write a few paragraphs about our new file formats (kind of what we have in commitlog.md) and then link to that documentation at the top of this page.

Otherwise as someone unfamiliar with the codebase its pretty difficult to decipher the what / why from code that is just manipulating bytes on disk if that makes sense.

Reply:
Yup sounds good, I'll write a `storage.md` and reference it from this module.  And maybe a short paragraph in the source file itself.

Reply:
+1, I found https://code.uberinternal.com/P79869 helpful while reviewing the code

Reply:
Ha thanks for that, removed the URL from your comment though to avoid leakage of non-public assets.

Reply:
lol cheers